spinta/manifests/xsd2/helpers.py

```diff
@@ -450,6 +469,75 @@ def _post_process_refs(self) -> None:
                     logger.warning(f"Parent model '{model.extends_model}' not found for model '{model.name}'")
                     model.extends_model = None
 
+    def _post_process_keys(self) -> None:
+        """
+        Apply xs:key constraints to models by setting the ref column.
+
+        The ref column indicates which properties form the primary key.
+        For composite keys, multiple property names are stored in a list.
+        """
+        for element_name, keys in self.keys.items():
+            # Find the model(s) for this element
+            if element_name not in self.top_level_element_models:
+                logger.warning(f"xs:key defined for unknown element: {element_name}")
+                continue
+
+            for key in keys:
+                # Parse selector to find target model
+                # Selector xpath like "City" or "Cities/City"
+                selector_parts = key.selector_xpath.split("/")
+                # Remove namespace prefixes (e.g., "tns:City" -> "City")
+                selector_parts = [part.split(":")[-1] for part in selector_parts]
+
+                # Get the target element name from selector
+                target_element_name = selector_parts[-1]
+
+                # Find target model(s) by xsd_name
+                # First check top_level_element_models (for root elements)
+                if target_element_name in self.top_level_element_models:
+                    target_models = self.top_level_element_models[target_element_name]
+                else:
+                    # Search through all models for nested elements
+                    target_models = [m for m in self.models if m.xsd_name == target_element_name]
+
+                if not target_models:
+                    logger.warning(f"xs:key selector references unknown element: {target_element_name}")
+                    continue
+
+                # Apply key to target model(s)
+                for target_model in target_models:
+                    # Parse field xpaths to property names
+                    ref_properties = []
+                    for field_xpath in key.field_xpaths:
+                        # Remove namespace prefix
+                        field_xpath = field_xpath.split(":")[-1]
+
+                        # Handle attribute vs element
+                        if field_xpath.startswith("@"):
+                            # Attribute: @ID -> find property with source="@ID"
+                            attr_name = field_xpath[1:]
+                            source_to_find = f"@{attr_name}"
+                        else:
+                            # Element: Name -> find property with source="Name/text()"
+                            source_to_find = f"{field_xpath}/text()"
+
+                        # Find matching property
+                        prop_name = None
+                        for pname, prop in target_model.properties.items():
```

Comment:
More explicit names could be given and abbreviations for variable names should be avoided. E.g. `prop` is a separate word in itself and has a [completely different meaning](https://www.google.com/search?q=prop+meaning&oq=prop+meaning&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIGCAEQLhhA0gEIMTUyNGowajGoAgCwAgA&sourceid=chrome&ie=UTF-8).

```suggestion
                        matching_property_name = None
                        for property_name, property in target_model.properties.items():
```